backend/core/src/secondary-adapters/postgres/channel/channel-repository-adapter.ts

```diff
@@ -11,6 +11,107 @@ export class ChannelRepositoryAdapter implements ChannelRepository {
     this.prismaClient = prismaClient;
   }
 
+  async addSubscription(userId: string, channelId: string): Promise<CreateSubscriptionResponseDto | null> {
+    await this.prismaClient.channel.update({
+      where: {
+        id: channelId,
+      },
+      data: {
+        subscriptions: {
+          createMany: {
+            data: {
+              userId,
+            },
+          },
+        },
+      },
+      select: {
+        subscriptions: {
+          where: { userId },
+        },
+      },
+    });
+
+    return { isSubscribe: true };
+  }
+
+  async removeSubscription(userId: string, channelId: string): Promise<CreateSubscriptionResponseDto | null> {
+    await this.prismaClient.channel.update({
+      where: {
+        id: channelId,
+      },
+      data: {
+        subscriptions: {
+          deleteMany: [{ userId }],
```

Comment:
why deleteMany?

Reply:
because of prisma documentation in this case we need to update channel by delete spesific related field subscribtion 
![image](https://user-images.githubusercontent.com/77640910/185999370-ddbcce52-da9e-4d34-a731-e504da37618c.png)